src/reporter/pegasus_counter_reporter.cpp

```diff
@@ -2,28 +2,47 @@
 // This source code is licensed under the Apache License Version 2.0, which
 // can be found in the LICENSE file in the root directory of this source tree.
 
-#include "pegasus_counter_reporter.h"
-
 #include <regex>
 #include <ios>
 #include <iomanip>
 #include <iostream>
 #include <unistd.h>
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <event2/event.h>
+#include <event2/buffer.h>
+#include <event2/http.h>
+#include <event2/keyvalq_struct.h>
 
 #include <dsn/cpp/service_app.h>
 #include <dsn/dist/replication/duplication_common.h>
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/flags.h>
 
 #include "base/pegasus_utils.h"
 #include "pegasus_io_service.h"
-
-#include <chrono>
-#include <map>
-#include <memory>
-#include <string>
-#include <dsn/dist/fmt_logging.h>
+#include "pegasus_counter_reporter.h"
 
 using namespace ::dsn;
 
+DSN_DEFINE_uint64("pegasus.server", prometheus_port, 9091, "prometheus exposer port");
+
+DSN_DEFINE_string("pegasus.server", falcon_host, "127.0.0.1", "falcon agent host");
+DSN_DEFINE_uint64("pegasus.server", falcon_port, 1988, "falcon agent port");
+DSN_DEFINE_string("pegasus.server", falcon_path, "/v1/push", "falcon http path");
+
+DSN_DEFINE_uint64("pegasus.server",
+                  perf_counter_update_interval_seconds,
+                  10,
+                  "perf_counter_update_interval_seconds");
+DSN_DEFINE_bool("pegasus.server", perf_counter_enable_logging, true, "perf_counter_enable_logging");
+DSN_DEFINE_string("pegasus.server", perf_counter_sink, "", "perf_counter_sink");
```

Comment:
Reorder as top-down, or dict order by config key would be better.

Reply:
I don't get your point. What's top-down order by config-key? Could you please write a suggestion showing that?

Reply:
There are 3 types of config group, top-down order is:
- configs for server            # top level, we can select prometheus or falcon
- configs for prometheus  # lower level
- configs for falcon            # lower level
So when developers read code here, he know it's useless to config falcon items when select prometheus sink.